cmd/frostfs-adm/internal/modules/morph/group.go

```diff
@@ -49,16 +49,15 @@ func initializeContractWallet(v *viper.Viper, walletDir string) (*wallet.Wallet,
 	return w, nil
 }
 
-func openContractWallet(v *viper.Viper, cmd *cobra.Command, walletDir string) (*wallet.Wallet, error) {
+func openContractWallet(v *viper.Viper, cmd *cobra.Command, walletDir string, createIfNotExists bool) (*wallet.Wallet, error) {
 	p := filepath.Join(walletDir, contractWalletFilename)
 	w, err := wallet.NewWalletFromFile(p)
 	if err != nil {
-		if !os.IsNotExist(err) {
-			return nil, fmt.Errorf("can't open wallet: %w", err)
+		if os.IsNotExist(err) && createIfNotExists {
+			cmd.Printf("Contract group wallet is missing, initialize at %s\n", p)
+			return initializeContractWallet(v, walletDir)
 		}
-
-		cmd.Printf("Contract group wallet is missing, initialize at %s\n", p)
-		return initializeContractWallet(v, walletDir)
+		return nil, fmt.Errorf("can't open wallet: %w", err)
```

Comment:
`set-config` _should_ still work if contract wallet does not exist, it isn't used in any way in it.
Does it work after this change or am I missing something?

Reply:
fixed